### ISSUE TITLE
Tainted but idle workers trend to 100% CPU and fail within a few hours

### ISSUE THREAD

**Issue by whereisaaron** [coreos/kube-aws#224]
So I created a test cluster with 0 workers and in the main pool, a regular pool of 2 workers, and a spot pool of 3 workers. The 3 spot pool workers were tainted and I scheduled nothing on them. When I came back the next day, all the spot instances had flipped to 'NotReady', two can't be pinged any more and the third I can ping but not ssh to (times out). Checking the monitoring, all three have their CPU pegged at 100%.
```
NAME                                        STATUS                     AGE        
ip-172-31-106-46.region.compute.internal    Ready,SchedulingDisabled   1d
ip-172-31-107-195.region.compute.internal   Ready                      1d
ip-172-31-107-39.region.compute.internal    Ready                      1d
ip-172-31-108-131.region.compute.internal   NotReady                   1d
ip-172-31-108-16.region.compute.internal    NotReady                   1d
ip-172-31-108-68.region.compute.internal    NotReady                   1d
```

This is similar to the problem if you run the `untaint-and-cordon` task on the controller (discussed in #199). The task is logged as failed (even if it succeeds) as (`rkt rm` fails), so the task repeats continuously. Some sort of garbage/clean-up accumulates and the controller eventually hits 100% CPU and grinds itself to death. After patching out the `untaint-and-cordon` task on my controllers I no longer have this problem (on the controllers).

It seem like any workers might have the same problem if you use the `experimental.taints` option. This is what I used on my spot instances:
```
experimental:
  taints:
    - key: lifecycle
      value: spot
      effect: NoSchedule
```

Here is the CPU history of the now unpingable tainted spot workers:

![dead-worker](https://cloud.githubusercontent.com/assets/1109585/21794572/7d36078a-d6c9-11e6-8d51-6e4fe262db0e.png)

![dead-m4](https://cloud.githubusercontent.com/assets/1109585/21794708/cff6d1e2-d6ca-11e6-9b0c-20d0f75e5796.png)

**Comment by mumoshu** [comment]
@whereisaaron Thanks again for your continuous feedbacks.

It definitely seems like an issue and I'd like it to be fixed in the same way as we've agreed in https://github.com/coreos/kube-aws/pull/199#issuecomment-271503071.

Would you mind sending a pull request for it or should I?

**Comment by redbaron** [comment]
What about keeping `rkt` but ignoring `rkt rm` errors? I am doing that in https://github.com/coreos/kube-aws/pull/217 

This would not trigger tasks restart and therefore no garbage will be accumulated with some "undeletable" garbage left until rkt bug is fixed.

I think ignoring `rkt rm` errors is fine as it is a cleanup task and main goal of the script already achieved.

**Comment by mumoshu** [comment]
@redbaron Thanks for your suggestion. Makes sense.

I'd personally prefer `docker run` w/ a comment linking to the rkt issue and TODO to migrate to rkt someday because it's a bit cleaner IMHO. However, I'm equally ok with `rkt run` followed `rkt rm` with ignoring errors on `rkt rm`. You can keep your PR as is in that regard 👍

**Comment by whereisaaron** [comment]
I prepared a PR #231 is switch to using docker to perform the taint task, at least until rkt is fixed. The exact same patch approach as used to fixed the same fault on the controller task in #199.

I tested deploying a node-pool or workers with this patch and the experimental.taint setting enabled and the right thing happened. The taint was applied and the node uncordoned. All looks good in the log and CPU load is nominal.

**Comment by lucab** [comment]
Partly hijacking the ticket for a rkt question: from the mounts number in the related PR, there seems to be a mount propagation loop that I would not expect from pods not using stage1-fly and performing proper rm. Can anybody please check and separately post if/what are the mounts being propagated and if there is a crash-looping fly pod?

**Comment by whereisaaron** [comment]
Hi @lucab have a read of #199 for more info. A failure to clean up mounts in `rkt rm` causes systemd units to loop and thus leak lots of mounts very quickly. That is fixed for now by using docker for the same task until `rkt rm` is fixed.

**Comment by mumoshu** [comment]
@lucab Would you mind sharing us how to look for/investigate into the mounts actually being over-propagated? Example commands would greatly help.

**Comment by mumoshu** [comment]
I'm closing this  as v0.9.3-rc.4 including the fix for this issue is released.
However please feel free to reopen if the issue still persists! Thanks.

**Comment by lucab** [comment]
@mumoshu `/proc/1/mountinfo` will show which mounts are being propagated and where. I think the root issue here is not caused by `taint-and-uncordon`, but by one of the fly pod.

@whereisaaron a fix is already shipped in rkt 1.22.0, released last week.
